### PR TITLE
feat: Add allowed_roles_list to oauth_integration_for_custom_clients

### PR DESCRIPTION
**Pull request by sfc-gh-kwasilewski**
Surfaces the ALLOWED_ROLES_LIST Snowflake property in the resource schema. Includes CustomizeDiff validation that the field requires oauth_use_secondary_roles = NONE, Read/Create/Update wiring, schema/describe changes, example and doc updates, and acceptance test coverage.

Adding allowed_roles_list to
snowflake_oauth_integration_for_partner_applications will be done in a follow-up.

References #4773

### CODE REVIEW

File: pkg/resources/oauth_integration_for_custom_clients.go [hunk 1]
```diff
@@ -82,6 +82,15 @@ var oauthIntegrationForCustomClientsSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: relatedResourceDescription("A set of Snowflake roles that a user does not need to explicitly consent to using after authenticating.", resources.AccountRole),
 	},
+	"allowed_roles_list": {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/oauth_integration_for_custom_clients.go:85]: question: in external_oauth_integration allowed_roles_list conflicts with blocked_roles_list - as I understand, it's not a problem here? at least based on the tests added

Comment (sfc-gh-asawicki) [inline pkg/resources/oauth_integration_for_custom_clients.go:85]: second question: what happens after the migration to this version when:
- the integration had some roles set up in Snowflake
- the integration had empty roles in Snowflake?

Asking because in the first case, the plan will not be empty, so customer should add it to their config then, right?

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/resources/oauth_integration_for_custom_clients.go (hunk 1 above)
Comment (sfc-gh-kwasilewski) [reply pkg/resources/oauth_integration_for_custom_clients.go:85]: No, both allowed and blocked can be set at the same time. The only caveat is that if a specific role is in ALLOWED_ROLES_LIST, it cannot be added to BLOCKED_ROLES_LIST.

Comment (sfc-gh-kwasilewski) [reply pkg/resources/oauth_integration_for_custom_clients.go:85]: 1. We detect a drift, and the customer should adjust their config. Should we mention that in the migration guide? Right now, it says "No action is required", but that may not be true.
2. The plan should be empty. I think it's already covered by the BasicUseCase test, where we create an integration with empty roles, and then we import it.

Comment (sfc-gh-asawicki) [reply pkg/resources/oauth_integration_for_custom_clients.go:85]: 2 is OK
ad 1 We should mention it, and also have a test confirming it will really happen (we had a situation where our recommendation was not aligned with what really happens). There is also an edge case, when the customer imports it (using import block which behaves a bit differently than terraform import command - sharing a short doc internally). For that reason, I'm not sure if we maybe shouldnt hide this behavior behind experiment.

Let's add a test and update migration guide entry and discuss on Monday if we need to add an experiment, ok?

Comment (sfc-gh-kwasilewski) [reply pkg/resources/oauth_integration_for_custom_clients.go:85]: Added test and adjusted migration guide

**Review by sfc-gh-asawicki — DISMISSED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]
